app/services/CartServiceImpl.java

```diff
@@ -123,16 +141,27 @@ public int getFrequency(final String cartId) {
 
     @Override
     public Cart createCartWithPactasInfo(final ProductProjection product, final PactasContract contract, final PactasCustomer customer) {
-        final ProductVariant variant = getVariantInContract(product, contract);
-        final Cart cart = sphereClient().execute(CartCreateCommand.of(CartDraft.of(DefaultCurrencyUnits.EUR))).toCompletableFuture().join();
-        Logger.debug("Created new Cart[cartId={}] with Pactas info", cart.getId());
-
-        final AddLineItem action = AddLineItem.of(product.getId(), variant.getId(), 1);
-        final Cart updatedCart = sphereClient().execute(CartUpdateCommand.of(cart, action)).toCompletableFuture().join();
-
-        final Address address = AddressBuilder.of(customer.getCompleteAddress()).build();
-        final Cart cartWithAddress = sphereClient().execute(CartUpdateCommand.of(updatedCart, SetShippingAddress.of(address))).toCompletableFuture().join();
-        return cartWithAddress;
+        return Optional.of(sphereClient().execute(CartCreateCommand.of(CartDraft.of(DefaultCurrencyUnits.EUR))).toCompletableFuture().join())
```

Comment:
You don't need to wrap the result in an optional. The CartCreateCommand always returns sth. non-null in the happy path and throws an exception in the error path.  Normally try-catch would be appropropriate for error wrapping. But with promises (a next step) this will change.